src/main/include/log4cxx/filter/mapfilter.h

```diff
@@ -19,26 +19,87 @@
 
 #include <log4cxx/spi/filter.h>
 
+#if defined(_MSC_VER)
+	#pragma warning ( push )
+	#pragma warning ( disable: 4251 )
+#endif
+
 namespace log4cxx
 {
 namespace filter
 {
+/**
+ * A Filter that operates on a Map.
+ */
 
+using KeyValList = std::map<LogString, LogString>;
```

Comment:
Using 'using' inside of a header is generally bad practice; in this case though, while it may be okay I don't see the point of creating a new type(this is equivalent to `typedef KeyValList std::map<LogString,LogString>`).

Reply:
no problem; I switch over to how its done in the existing propertyfilter.h